transmission/Dockerfile

```diff
@@ -7,12 +7,34 @@ COPY rootfs /
 
 # Setup base
 RUN apk add --no-cache \
-    coreutils=8.29-r2 \
+    coreutils \
     transmission-daemon \
     openvpn
+RUN apk add --no-cache \
+    nginx=1.14.2-r0
 
-# Scrip to run after startup
-CMD ["/usr/bin/run.sh"]
+# Small hack needed for ingress support
+#
+# Transmission always uses "transmission" as a subdirectory in the URL, so the web interface can for instance be found at:
+#
+#   http://<server>/transmission/web/
+#
+# The same goes for the RPC API, which is exposed under /transmission/rpc. When using ingress, this is not the case however
+# and "transmission" directory is not part of the URL. So transmission is basically served under the root (/), e.g.:
+#
+#   http://10.0.10.110:8123/api/hassio_ingress/rQEi5X4zJFTmZ82ghplDjsx2AoK3r2lqx0zZHHjZ69Q/
+#
+# As the web UI uses a relative path when resolving the RPC endpoint, it basically becomes:
+#
+#   http://10.0.10.110:8123/api/hassio_ingress/rQEi5X4zJFTmZ82ghplDjsx2AoK3r2lqx0zZHHjZ69Q/../rpc
+#
+# which obviously is:
+#
+#   http://10.0.10.110:8123/api/hassio_ingress
+#
+# This will not work. The sed statement below changes the way the RPC endpoint is determined by keeping the regular endpoint
+# when accessing transmission as usual, but stripping the relative part when using ingress.
+RUN sed -i 's|'"'"'\.\./rpc'"'"'|window.location.pathname.endsWith("/web/") ? "../rpc" : "rpc"|' /usr/share/transmission/web/javascript/remote.js
```

Comment:
Can't you do this less hacky with an nginx `sub_filter`?

`sub_filter '../rpc' 'rpc';` or something like that in nginx.conf.

Reply:
That was my intention at first, but I don't think that is possible. The RPC endpoint address is determined with javascript in the user browser, which means that the URL must end with something like `http://.../transmission/web/` for the web interface to work. So the correct endpoint is translated to `http://.../transmission/web/../rpc` -> `http://.../transmission/rpc`. The name "transmission" is of course irrelevant, it just has to be a subdirectory.

Since Home Assistant acts as a proxy to the addon (i.e. to nginx with in turn talks to transmission), it determines the "base url" the browser sees and it doesn't really matter what I do in nginx. I can't propagate rewrites, or whatever back to Home Assistant to gain that additional subdirectory.

Feel free to prove me wrong 😄